toolkits/confluence/arcade_confluence/client.py

```diff
@@ -0,0 +1,535 @@
+            await self.get_space_by_id(space_identifier)
+            if space_identifier.isdigit()
+            else await self.get_space_by_key(space_identifier)
+        )
+
+    async def get_page_id(self, page_identifier: str) -> str:
+        """Get the ID of a page from its identifier.
+
+        Args:
+            page_identifier: The identifier of the page to get. Can be a page's ID or title.
+
+        Returns:
+            The ID of the page
+        """
+        if page_identifier.isdigit():
+            page_id = page_identifier
+        else:
+            page = await self.get_page_by_title(page_identifier)
+            page_id = page.get("page", {}).get("id")
+        return page_id
+
+    async def get_space_id(self, space_identifier: str) -> str:
+        """Get the ID of a space from its identifier.
+
+        Args:
+            space_identifier: The identifier of the space to get. Can be a space's ID or title.
+        """
+        if space_identifier.isdigit():
+            space_id = space_identifier
+        else:
+            space = await self.get_space_by_key(space_identifier)
+            space_id = space.get("space", {}).get("id")
+        return space_id
+
+    def create_space_tree(self, space: dict) -> dict:
+        """Create the initial tree structure for a space.
+
+        Args:
+            space: The transformed space object
+
+        Returns:
+            A dictionary representing the root of the space hierarchy tree without any children
+        """
+        space_internal = space.get("space", {})
+        return {
+            "key": space_internal.get("key"),
+            "id": space_internal.get("id"),
+            "type": "space",
+            "url": space_internal.get("url"),
+            "children": [],
+        }
+
+    def convert_root_pages_to_tree_nodes(self, pages: list) -> list:
+        """Convert root pages of a space to tree nodes.
+
+        Args:
+            pages: List of page objects from the API
+
+        Returns:
+            A list of tree nodes representing the root pages
+        """
+        return [
+            {
+                "title": page.get("title"),
+                "id": page.get("id"),
+                "type": "page",
+                "url": page.get("url"),
+                "children": [],
+            }
+            for page in pages
+        ]
+
+    async def process_page_descendants(self, root_children: list, base_url: str) -> None:
+        """Process descendants for each page and build the hierarchy.
+
+        Args:
+            root_children: The root children of the space
+            base_url: The base URL for the Confluence space
+
+        Returns:
+            None (modifies root_children in place)
+        """
+        descendent_params = {"limit": 250, "depth": 5}
+
+        for i, child in enumerate(root_children):
+            page_id = child["id"]
+            descendants = await self.get(f"pages/{page_id}/descendants", params=descendent_params)
```

Comment:
Since the number of children could be somewhat large, it'd be interesting to take advantage of async and retrieve multiple at a time here to speed up the process.

Might be desirable to have some sort of timeout control here as well, to prevent the tool from running for too long in the case of large Confluence spaces...